src/features/completion.ts

```diff
@@ -837,19 +837,50 @@ export class JSONCompletion {
     if (
       !subSchema ||
       subSchema.name === "UnknownPropertyError" ||
+      // TODO: Looking into why this is needed
       subSchema.enum ||
       subSchema.type === "undefined" ||
       subSchema.type === "null"
     ) {
+      debug.log(
+        "xxx",
+        "no schema for pointer. Try parent pointer..",
+        pointer,
+        subSchema
+      );
       pointer = pointer.replace(/\/[^/]*$/, "/");
-      subSchema = draft.getSchema({ pointer });
+      pointer = "/" ? "" : pointer; // root pointer should be empty ("/" is a JSON pointer for an empty string key)
+      subSchema = draft.getSchema({ pointer, data: data ?? undefined });
+      debug.log(
+        "xxx",
+        "subSchema for parent pointer",
+        subSchema,
+        pointer,
+        data
+      );
+      // resolve all direct child schemas
+      // TODO: This is a bit hacky, but it works for now
+      if (subSchema?.properties) {
+        Object.entries(subSchema.properties).forEach(([key, value]) => {
+          if (subSchema?.properties && typeof value === "object") {
```

Comment:
is there a reason to check for subSchema.properties twice here?

Reply:
typescript typechecks doesn't figure out that it has already been checked since this is an internal function